src/Symfony/Component/Security/Core/Authorization/Voter/Voter.php

```diff
@@ -48,24 +49,56 @@ public function vote(TokenInterface $token, $subject, array $attributes)
             }
 
             // as soon as at least one attribute is supported, default is to deny access
-            $vote = self::ACCESS_DENIED;
+            $vote = $this->deny();
 
-            if ($this->voteOnAttribute($attribute, $subject, $token)) {
+            $decision = $this->voteOnAttribute($attribute, $subject, $token);
+            if (\is_bool($decision)) {
+                trigger_deprecation('symfony/security-core', '5.3', 'Returning a boolean in "%s::voteOnAttribute()" is deprecated, return an instance of "%s" instead.', static::class, Vote::class);
+                $decision = $decision ? $this->grant() : $this->deny();
+            }
+
+            if ($decision->isGranted()) {
                 // grant access as soon as at least one attribute returns a positive response
-                return self::ACCESS_GRANTED;
+                return $decision;
             }
+
+            $vote->setReason($vote->getReason().trim(' '.$decision->getReason()));
         }
 
         return $vote;
     }
 
+    /**
+     * Creates a granted vote.
+     */
+    public function grant(string $reason = '', array $parameters = []): Vote
+    {
+        return Vote::createGranted($reason, $parameters);
+    }
+
+    /**
+     * Creates an abstained vote.
+     */
+    public function abstain(string $reason = '', array $parameters = []): Vote
+    {
+        return Vote::createAbstain($reason, $parameters);
+    }
+
+    /**
+     * Creates a denied vote.
+     */
+    public function deny(string $reason = '', array $parameters = []): Vote
+    {
+        return Vote::createDenied($reason, $parameters);
+    }
+
     /**
      * Determines if the attribute and subject are supported by this voter.
      *
      * @param string $attribute An attribute
      * @param mixed  $subject   The subject to secure, e.g. an object the user wants to access or any other PHP type
      *
-     * @return bool
+     * @return bool True if the attribute and subject are supported, false otherwise
```

Comment:
this does not provide benefits. The description of the function already explains what `supported` means, and I guess nobody will have the idea that we might return `false` as the boolean representing supported things.